submissions/tests/test_api.py

```diff
@@ -268,6 +271,28 @@ def test_set_score_signal(self, send_mock):
             item_id=STUDENT_ITEM['item_id']
         )
 
+    @ddt.data(u"First score was incorrect", u"☃")
+    def test_set_score_with_annotation(self, reason):
+        submission = api.create_submission(STUDENT_ITEM, ANSWER_ONE)
+        creator_uuid = "Bob"
+        annotation_type = "staff_override"
+        api.set_score(submission["uuid"], 11, 12, creator_uuid, annotation_type, reason)
+        score = api.get_latest_score_for_submission(submission["uuid"])
+        self._assert_score(score, 11, 12)
+
+        # We need to do this to verify that one score annotation exists and was
+        # created for this score. We do not have an api point for retrieving
+        # annotations, and it doesn't make sense to expose them, since they're
+        # for auditing purposes.
+        annotations = ScoreAnnotation.objects.all()
+        self.assertGreater(len(annotations), 0)
+        annotation = annotations[0]
+        self.assertEqual(annotation.score.points_earned, 11)
+        self.assertEqual(annotation.score.points_possible, 12)
+        self.assertEqual(annotation.annotation_type, annotation_type)
+        self.assertEqual(annotation.creator, creator_uuid)
+        self.assertEqual(annotation.reason, reason)
+
```

Comment:
I'd add a test where the score annotation is invalid (say the annotation_type is too long) and verify the behavior.